internal/predicate/nonadminbackup_predicate.go

```diff
@@ -19,86 +19,40 @@ package predicate
 import (
 	"context"
 
-	"github.com/go-logr/logr"
-	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/event"
-	"sigs.k8s.io/controller-runtime/pkg/log"
 
-	nacv1alpha1 "github.com/migtools/oadp-non-admin/api/v1alpha1"
-	"github.com/migtools/oadp-non-admin/internal/common/constant"
+	"github.com/migtools/oadp-non-admin/internal/common/function"
 )
 
-// NonAdminBackupPredicate contains event filters for Non Admin Backup objects
-type NonAdminBackupPredicate struct {
-	Logger logr.Logger
-}
+const nonAdminBackupPredicateKey = "NonAdminBackupPredicate"
 
-func getNonAdminBackupPredicateLogger(ctx context.Context, name, namespace string) logr.Logger {
-	return log.FromContext(ctx).WithValues("NonAdminBackupPredicate", types.NamespacedName{Name: name, Namespace: namespace})
-}
+// NonAdminBackupPredicate contains event filters for Non Admin Backup objects
+type NonAdminBackupPredicate struct{}
 
```

Comment:
just add small concise comments regarding Predicate functions per events so that we dont forget the decisions we made here.
Like for `Create` We are accepting all kinds of NAB create events here.
For `Update` We are only reconciling on spec changes and so on.
It will really be helpful in the long run.

Reply:
will be applied in next commit